irohad/network/impl/async_grpc_client.hpp

```diff
@@ -32,7 +32,9 @@ namespace iroha {
     template <typename Response>
     class AsyncGrpcClient {
      public:
-      AsyncGrpcClient() : thread_(&AsyncGrpcClient::asyncCompleteRpc, this) {}
+      AsyncGrpcClient(logger::Logger &&log)
+          : thread_(&AsyncGrpcClient::asyncCompleteRpc, this),
+            log_(std::move(log)) {}
```

Comment:
Why do you call `std::move` on rvalue ref?

Reply:
because of [c++ typing magic](https://ideone.com/31oVDm)